lib/engine-addon.js

```diff
@@ -406,21 +476,15 @@ module.exports = {
         // LAZY LOADING!
         // But we have to implement everything manually for the lazy loading scenario.
 
-        // Get base styles tree.
-        var engineStylesTree = buildEngineCSSTree.call(this);
+        var vendorTree = buildVendorTree.call(this);
+        var vendorJSTree = buildVendorJSTree.call(this, vendorTree);
+        var externalTree = new Funnel(vendorTree, {
+          srcDir: 'vendor',
+          destDir: 'vendor',
+          allowEmpty: true
+        });
 
-        // Move styles tree into the correct place.
-        // `**/*.css` all gets merged.
-        // The addon.css file has already been renamed to match `this.name`.
-        // All we need to do is concatenate it down.
-        var primaryStyleTree;
-        if (engineStylesTree) {
-          primaryStyleTree = concat(engineStylesTree, {
-            allowNone: true,
-            inputFiles: ['**/*.css'],
-            outputFile: 'engines-dist/' + this.name + '/assets/engine.css'
-          });
-        }
+        var finalStylesTree = this.compileLazyEngineStyles(vendorTree, externalTree);
```

Comment:
I'm concerned that this doesn't prevent hoisting of nested lazy engine's styles but haven't traced through it. I believe that this case isn't covered in our fixtures.

Given: `root > lazy1 > lazy2 > eager`

We would expect:
- `root.css` to have only its own assets.
- `lazy1/engine-vendor.css` to be empty.
- `lazy2/engine-vendor.css` to contain `eager`'s styles.